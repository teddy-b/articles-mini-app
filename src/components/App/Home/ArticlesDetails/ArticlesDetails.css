.articles-details {
  padding: 0 1em;
  width: 60%;
}

.articles-details .articles-details-empty {
  width: 100%;
  text-align: center;
}
