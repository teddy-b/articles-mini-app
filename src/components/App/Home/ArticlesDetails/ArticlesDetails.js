import React from 'react';

import Article from './Article';
import CommentsList from './CommentsList';
import './ArticlesDetails.css';

const ArticlesDetails = ({
  comments,
  onAddComment,
  onAddReply,
  onLoadReplies,
  selectedArticle
}) => (
  selectedArticle
    ? (
      <div className="articles-details">
        <Article {...selectedArticle} />
        <CommentsList
          comments={comments}
          onAddComment={onAddComment}
          onAddReply={onAddReply}
          onLoadReplies={onLoadReplies}
          selectedArticleId={selectedArticle.id}
        />
      </div>
    )
    : (
      <div className="articles-details-empty">
        <h2>Nothing to show</h2>
        <div>Select article from the list to see details</div>
      </div>
    )
);

export default ArticlesDetails;
