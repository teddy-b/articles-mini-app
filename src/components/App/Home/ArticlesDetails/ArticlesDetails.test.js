import React from 'react';
import renderer from 'react-test-renderer';

import ArticlesDetails from './ArticlesDetails';

jest.mock('./Article', () => () => 'Article');
jest.mock('./CommentsList', () => 'CommentsList');

describe('ArticlesDetails', () => {
  const props = {
    comments: [],
    onAddComment: jest.fn(),
    onAddReply: jest.fn(),
    onLoadReplies: jest.fn(),
    selectedArticle: null
  };

  it('renders correctly', () => {
    const component = renderer.create(<ArticlesDetails {...props} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with selected article', () => {
    const component = renderer.create(
      <ArticlesDetails
        {...props}
        selectedArticle={1}
      />
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
