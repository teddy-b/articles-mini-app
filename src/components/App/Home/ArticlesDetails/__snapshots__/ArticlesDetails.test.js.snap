// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticlesDetails renders correctly 1`] = `
<div
  className="articles-details-empty"
>
  <h2>
    Nothing to show
  </h2>
  <div>
    Select article from the list to see details
  </div>
</div>
`;

exports[`ArticlesDetails renders correctly with selected article 1`] = `
<div
  className="articles-details"
>
  Article
  <CommentsList
    comments={Array []}
    onAddComment={[Function]}
    onAddReply={[Function]}
    onLoadReplies={[Function]}
    selectedArticleId={undefined}
  />
</div>
`;
